### PR TITLE
1.x: Added MergeDelay operators for Iterable of Observables

### PR DESCRIPTION
**Pull request by JohnWowUs**
PR for issue #3577

### CODE REVIEW

**Comment by akarnokd** [comment]
:+1:

File: src/main/java/rx/Observable.java [hunk 1]
```diff
@@ -2186,6 +2186,65 @@ public void call(Subscriber<? super Object> subscriber) {
...
+     *  <dd>{@code mergeDelayError} does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     * 
+     * @param sequences
+     *            the Iterable of Observables
+     * @return an Observable that emits items that are the result of flattening the items emitted by the
+     *         Observables in the Iterable
+     * @see <a href="http://reactivex.io/documentation/operators/merge.html">ReactiveX operators documentation: Merge</a>
+     */
+    public final static <T> Observable<T> mergeDelayError(Iterable<? extends Observable<? extends T>> sequences) {
```
Comment (artem-zinnatullin) [inline src/main/java/rx/Observable.java]: Please remove `final`

File: src/main/java/rx/Observable.java [hunk 2]
```diff
@@ -2186,6 +2186,65 @@ public void call(Subscriber<? super Object> subscriber) {
...
+     * 
+     * @param sequences
+     *            the Iterable of Observables
+     * @param maxConcurrent
+     *            the maximum number of Observables that may be subscribed to concurrently
+     * @return an Observable that emits items that are the result of flattening the items emitted by the
+     *         Observables in the Iterable
+     * @see <a href="http://reactivex.io/documentation/operators/merge.html">ReactiveX operators documentation: Merge</a>
+     */
+    public final static <T> Observable<T> mergeDelayError(Iterable<? extends Observable<? extends T>> sequences, int maxConcurrent) {
```
Comment (artem-zinnatullin) [inline src/main/java/rx/Observable.java]: `final` is not needed here

File: src/test/java/rx/internal/operators/OperatorMergeDelayErrorTest.java [hunk 3]
```diff
@@ -272,6 +272,23 @@ public void testMergeList() {
         verify(stringObserver, times(2)).onNext("hello");
     }
 
+    // This is pretty much a clone of testMergeList but with the overloaded MergeDelayError for Iterables
+    @Test     
+    public void testMergeIterable() {
```
Comment (artem-zinnatullin) [inline src/test/java/rx/internal/operators/OperatorMergeDelayErrorTest.java]: `test` prefix is not needed here, `@Test` does the work

**Comment by artem-zinnatullin** [comment]
Just few things about code style, otherwise LGTM :+1:

File: src/test/java/rx/internal/operators/OperatorMergeDelayErrorTest.java (hunk 3 above)
Comment (JohnWowUs) [inline src/test/java/rx/internal/operators/OperatorMergeDelayErrorTest.java]: True but added a test prefix seems to be the house style.

File: src/test/java/rx/internal/operators/OperatorMergeDelayErrorTest.java [hunk 4]
```diff
@@ -272,6 +272,23 @@ public void testMergeList() {
         verify(stringObserver, times(2)).onNext("hello");
     }
 
+    // This is pretty much a clone of testMergeList but with the overloaded MergeDelayError for Iterables
+    @Test     
+    public void MergeIterable() {
```
Comment (zsxwing) [inline src/test/java/rx/internal/operators/OperatorMergeDelayErrorTest.java]: nit: **M**ergeIterable -> **m**ergeIterable

**Comment by zsxwing** [comment]
:+1: once you fix the style issue

**Comment by stevegury** [comment]
:+1: after renaming the `mergeIterable` method.

**Comment by zsxwing** [comment]
:+1: @JohnWowUs can you please squash commits in your PR into one?

**Comment by zsxwing** [comment]
@JohnWowUs could you squash commits? Thanks!

**Comment by JohnWowUs** [comment]
@zsxwing 

Done.

**Comment by zsxwing** [comment]
:+1:

**Pull request merged**